### PR TITLE
Add LuaMemberAttribute

### LINKED ISSUE #488: Add an attribute to allow developers to choose member names when registered in Lua
I've often ran into the issue of having instances of classes in C# that I want to make accessible in Lua, but I've always found it annoying that it registers with the exact same name as in C#, meaning that if you have the following:

```c#
class TestClass
{
    public int Value { get; set; }
    public string GetCoolString() { return "Cool String!"; }
}
```
which follows common C# naming conventions, and register it to be accessible in lua with `lua["test"] = new TestClass()`, then you can only access `Value` by `test.Value`, and `GetCoolString()` is accessed by `test:GetCoolString()`. This naming convention in Lua is not very however, and there should be ways for developers to conform to a Lua naming convention without having to name their things in C# in strange ways.

It would be nice to be able to do something like this instead (of course preserving the old way as well, just having this as an optional attribute that allows developers to choose a different name if they want to):

```c#
class TestClass
{
    [LuaMember(Name = "value")]
    public int Value { get; set; }
    [LuaMember(Name = "get_cool_string")]
    public string GetCoolString() { return "Cool String!"; }
}
```

which then would register `test.value` and `test:get_cool_string()` instead!

### PR DESCRIPTION
**Pull request by dcronqvist**
Closes #488 by adding an attribute to allow developers to select their own names for fields/properties/methods when they are registered in Lua.

### CODE REVIEW

File: src/LuaGlobals.cs [hunk 1]
```diff
@@ -177,11 +183,19 @@ private void RegisterClassOrInterface(string path, Type type, int recursionCount
             {
                 if (
                     // Check that the LuaHideAttribute and LuaGlobalAttribute were not applied
-                    (!field.GetCustomAttributes(typeof(LuaHideAttribute), false).Any()) &&
-                    (!field.GetCustomAttributes(typeof(LuaGlobalAttribute), false).Any()))
+                    (!field.GetCustomAttributes(typeof(LuaHideAttribute), false).Any()))
                 {
+                    var name = field.Name;
```
Comment (viniciusjarina) [inline src/LuaGlobals.cs:188]: Avoid `var` when the type is not explicit or for primitive types

Comment (dcronqvist) [reply src/LuaGlobals.cs]: Good catch! Fixed in several places :)

**Comment by viniciusjarina** [comment]
It seems iOS build is failing. NUnit on iOS is a bit more restricted than the desktop one 
```
"/Users/runner/work/1/s/NLua.Mac.sln" (default target) (1) ->
"/Users/runner/work/1/s/tests/build/xamarinios/NLuaTest.XamariniOS.csproj" (default target) (3) ->
(CoreCompile target) -> 
  /Users/runner/work/1/s/tests/src/LuaTests.cs(3255,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarinios/NLuaTest.XamariniOS.csproj]
  /Users/runner/work/1/s/tests/src/LuaTests.cs(3256,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarinios/NLuaTest.XamariniOS.csproj]
  /Users/runner/work/1/s/tests/src/LuaTests.cs(3257,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarinios/NLuaTest.XamariniOS.csproj]
  /Users/runner/work/1/s/tests/src/LuaTests.cs(3258,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarinios/NLuaTest.XamariniOS.csproj]
  /Users/runner/work/1/s/tests/src/LuaTests.cs(3261,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarinios/NLuaTest.XamariniOS.csproj]
  /Users/runner/work/1/s/tests/src/LuaTests.cs(3262,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarinios/NLuaTest.XamariniOS.csproj]


"/Users/runner/work/1/s/NLua.Mac.sln" (default target) (1) ->
"/Users/runner/work/1/s/tests/build/xamarintvos/NLuaTest.XamarinTVOS.csproj" (default target) (5) ->
  /Users/runner/work/1/s/tests/src/LuaTests.cs(3255,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarintvos/NLuaTest.XamarinTVOS.csproj]
  /Users/runner/work/1/s/tests/src/LuaTests.cs(3256,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarintvos/NLuaTest.XamarinTVOS.csproj]
  /Users/runner/work/1/s/tests/src/LuaTests.cs(3257,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarintvos/NLuaTest.XamarinTVOS.csproj]
  /Users/runner/work/1/s/tests/src/LuaTests.cs(3258,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarintvos/NLuaTest.XamarinTVOS.csproj]
  /Users/runner/work/1/s/tests/src/LuaTests.cs(3261,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarintvos/NLuaTest.XamarinTVOS.csproj]
  /Users/runner/work/1/s/tests/src/LuaTests.cs(3262,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarintvos/NLuaTest.XamarinTVOS.csproj]
```

**Comment by dcronqvist** [comment]
@viniciusjarina Changed to use IEnumerable<T>.Contains instead of relying on Assert.Contains, seems to have worked. But looks like there are other issues with the Windows build now. Something wrong on AzDevops or with the changes I've made?

**Comment by viniciusjarina** [comment]
@dcronqvist yeah. I bump the image to windows-2022 that is missing a UWP SDK, I need to bump the SDK version in the UWP project. I am doing this and merging your PR. As soon I get a green build I will trigger a release

**Pull request merged**